instagram-app/src/components/PostContainer/PostsPage.js

```diff
@@ -0,0 +1,98 @@
+import React, { Component } from "react";
+import Fuse from "fuse.js";
+import data from "../../dummy-data";
+import SeachBar from "../SearchBar/SearchBar";
+import PostContainer from "./PostContainer";
+
+export default class PostsPage extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      posts: [],
+      searchInput: ""
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener("beforeunload", () => {
```

Comment:
If add an event listener on componentDidMount is always a good idea to remove it on componentDidUnmount. Otherwise it's not garbage collected and if your component mount and unmount few times you end with performance propblems.